File: app/src/main/java/com/bovink/androidlearning/UserDao.java
package com.bovink.androidlearning;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

/**
 * @author bovink
 * @since 2021/1/12
 */
@Dao
public interface UserDao {
    @Query("SELECT * FROM user")
    List<User> getALl();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<User> users);

    @Delete
    void delete(User user);
}

File: app/src/main/java/com/bovink/androidlearning/UserDao.kt
package com.bovink.androidlearning

import androidx.room.*
import kotlinx.coroutines.flow.Flow

/**
 * @author bovink
 * @since 2021/6/9
 */
@Dao
interface UserDao {
    @Query("SELECT * FROM user")
    suspend fun getAll(): List<User>

    @Query("SELECT * FROM user")
    fun loadUsers(): Flow<List<User>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAll(vararg users: User)

    @Delete
    suspend fun delete(user: User)

}